api/pkg/models/user_devices.go

```diff
@@ -10,13 +10,14 @@ import (
 )
 
 type UserDevice struct {
-	ID        uint      `json:"id"         gorm:"primaryKey"`
-	User      User      `json:"user"`
-	UserID    uint      `json:"user_id"    gorm:"uniqueIndex:user_devices_user_id_device_id_key"`
-	Device    Device    `json:"device"`
-	DeviceID  uint      `json:"device_id"  gorm:"uniqueIndex:user_devices_user_id_device_id_key"`
-	CreatedAt time.Time `json:"created_at"`
```

Comment:
There was no need for a NewlyCreated field before. Why is it required now?

Reply:
It is no required, but it's the solution I found not to declare a variable (newDevices) in controller and pass its reference through 2 to 3 function (GetOrCreateUser). Also we would have had to pass newDevices every time we call GetOrCreateUser